### PR TITLE
Add Search Display of ACLs

### PR DESCRIPTION
**Pull request by seamuslee001**
Overview
----------------------------------------
This adds a search kit built search display of ACLs into the new admin ui extension and associated metadata that is needed and includes #26615 

Before
----------------------------------------
No new display for list of ACLs based on search kit

After
----------------------------------------
New search kit display

ping @colemanw

### CODE REVIEW

File: ext/civicrm_admin_ui/managed/SavedSearch_Manage_ACLs.mgd.php [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+              'label' => E::ts('Enabled'),
+              'sortable' => TRUE,
+            ],
+            [
+              'type' => 'field',
+              'key' => 'deny',
+              'dataType' => 'Boolean',
+              'label' => E::ts('Mode'),
+              'sortable' => TRUE,
+              'rewrite' => '{if "[deny]" eq "1"}Deny{else}Allow{/if}',
```
Comment (colemanw) [inline ext/civicrm_admin_ui/managed/SavedSearch_Manage_ACLs.mgd.php:118]:
```suggestion
              'rewrite' => '{if "[deny]" eq "1"}' . E::ts('Deny') . '{else}' . E::ts('Allow') . '{/if}',
```

File: CRM/ACL/BAO/ACL.php [hunk 2]
```diff
@@ -524,4 +526,45 @@ protected static function loadPermittedIDs(int $contactID, string $tableName, in
...
+      'name' => 0,
+    ];
+    $options = civicrm_api4($entity, 'get', [
+      'select' => [$label, 'id', 'name'],
+    ]);
+    foreach ($options as $option) {
+      $finalOptions[] = [
+        'label' => $option[$label],
+        'id' => $option['id'],
+        'name' => $option['name'],
```
Comment (colemanw) [inline CRM/ACL/BAO/ACL.php:564]: I guess not every entity has a name so:
```suggestion
        'name' => $option['name'] ?? $option['id'],
```

File: CRM/Core/DAO.php [hunk 3]
```diff
@@ -2776,17 +2776,17 @@ public static function appendPseudoConstantsToFields(&$fields) {
    * @param string $fieldName
    * @param string $context
    * @see CRM_Core_DAO::buildOptionsContext
-   * @param array $props
+   * @param array $values
    *   Raw field values; whatever is known about this bao object.
    *
    * Note: $props can contain unsanitized input and should not be passed directly to CRM_Core_PseudoConstant::get
```
Comment (colemanw) [inline CRM/Core/DAO.php:2782]:
```suggestion
   * Note: $values can contain unsanitized input and should be handled with care by CRM_Core_PseudoConstant::get
```

File: CRM/ACL/BAO/ACL.php [hunk 4]
```diff
@@ -524,4 +526,45 @@ protected static function loadPermittedIDs(int $contactID, string $tableName, in
...
+      'civicrm_custom_group' => ts('Custom Group'),
+    ];
+  }
+
+  public static function getObjectIdOptions($context, $params): array {
+    if (empty($params['values']['object_table']) && empty($params['values']['object_table:label'])) {
+      return [];
+    }
+    if (!empty($params['values']['object_table:label'])) {
+      $table_name = array_flip(self::operationTables())[$params['values']['object_table:label']];
```
Comment (colemanw) [inline CRM/ACL/BAO/ACL.php:543]:
```suggestion
      $table_name = array_flip(self:: getObjectTableOptions())[$params['values']['object_table:label']];
```

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by colemanw** [comment]
Hooray, passing test!

File: xml/schema/ACL/ACL.xml [hunk 5]
```diff
@@ -99,6 +105,10 @@
     <type>int unsigned</type>
     <comment>The ID of the object controlled by this ACL entry</comment>
     <add>1.6</add>
+    <pseudoconstant>
+      <callback>CRM_ACL_BAO_ACL::getObjectIdOptions</callback>
+      <prefetch>false</prefetch>
```
Comment (colemanw) [inline xml/schema/ACL/ACL.xml:110]: @seamuslee001 I just noticed this. Why did you add this `<prefetch>` flag? I'm honestly not even sure what this would do when paired with a `<callback>`, I don' think that's ever been done before.
